### PR TITLE
Add an online documentation site with GitHub Pages

### PR DESCRIPTION
**Pull request by j-f1**
- [x] I have read the [section in CONTRIBUTING.md about pull requests](https://github.com/electron-userland/electron-packager/blob/master/CONTRIBUTING.md#filing-pull-requests).

**Summarize your changes:**
This PR adds a GitHub pages site that displays the documentation.
Currently, there is no documentation index, so it redirects to `/api`.

See [the live demo](https://j-f1.github.io/electron-packager).

**Are your changes appropriately documented?**
**Do your changes have sufficient test coverage?**
**Does the testsuite pass successfully on your local machine?**
N/A

**TODO**

- [ ] Enable GitHub Pages
- [ ] Choose an appropriate Jekyll theme (anything that isn’t [listed here](https://pages.github.com/themes) needs further exploration)
- [x] Include the API docs
- [x] Link to the API docs from the README
- [x] Symlink the README to `docs/index.md`
- [ ] Squash Commits

### CODE REVIEW

**Comment by malept** [comment]
Thanks for the contribution, I think we'd want a better theme though. And possibly have a better index page, that was basically a mirror of the readme. _(I have no idea how to do that.)_

@MarshallOfSound we should probably figure out what a "Userland" theme is and apply it to both Packager and Forge websites.

**Comment by malept** [comment]
> Which theme would be better?

See my comment to @MarshallOfSound - basically it would be something similar (but I don't think it should be quite the same) to https://beta.electronforge.io/.

**Comment by malept** [comment]
Hmmm. I think you should add
```

[ci skip]
```

To your commits until they get squashed. I don't think this PR should trigger CI for now.

**Comment by j-f1** [comment]
Thanks!

**Comment by malept** [comment]
Oh, that's much nicer.

File: readme.md [hunk 1]
```diff
@@ -1,163 +0,0 @@
-# electron-packager
```
Comment (malept) [inline readme.md]: My concern with having the readme be a symlink is that IIRC symlinks do not work on Windows.

Comment (j-f1) [inline readme.md]: I tried having the symlinks be the other way, but then GitHub Pages wouldn’t build.

Comment (malept) [inline readme.md]: Not at the moment, unfortunately.

**Comment by malept** [comment]
Is renaming the readme really necessary?

**Comment by j-f1** [comment]
@malept No, I’ll revert that. I emailed GitHub support, and they told me that the only way to get this to work would be to have GitHub Pages target the root of the master branch or the `gh-pages` branch.

**Comment by malept** [comment]
Also, sorry about the merge conflict. I was messing with the GitHub Pages settings and I didn't realize that it would commit to the repository.

**Comment by j-f1** [comment]
@malept How would you like to go forward? We can:
* Move anything that should be public on the web to the `gh-pages` branch, or
* Make the entire `master` branch be on GitHub pages.

**Comment by malept** [comment]
I know it's a bit more work (and I might do it if you decline, which is a completely fine decision), but I'm thinking that Travis should (re)create the `gh-pages` branch on every master commit, similar to how [`travis-cargo`] works for Rust projects. That also means that we can use a custom theme.

**Comment by j-f1** [comment]
I’m closing this until I figure out the auto-generation.

**Pull request closed without merging**